common/djangoapps/third_party_auth/models.py

```diff
@@ -880,12 +900,34 @@ def get_config(self):
         conf['x509certMulti'] = {'signing': public_keys}
         conf['x509cert'] = ''
         conf['url'] = sso_url
+        # This block determines which SAML configuration should be used during authentication
+        if self.saml_configuration:
+            direct_config = self.saml_configuration
+            conf['saml_sp_configuration'] = direct_config
+            # .. custom_attribute_name: saml_config.use_latest_toggle_enabled
+            # .. custom_attribute_description: True if USE_LATEST_SAML_CONFIG is enabled; false otherwise.
+            set_custom_attribute('saml_config.use_latest_toggle_enabled', USE_LATEST_SAML_CONFIG.is_enabled())
+            if USE_LATEST_SAML_CONFIG.is_enabled():
+                try:
+                    slug = self.saml_configuration.slug
+                    site_id = self.saml_configuration.site_id
+                    latest_config = SAMLConfiguration.current(self.saml_configuration.site_id, self.saml_configuration.slug)
+                    if latest_config:
+                        # .. custom_attribute_name: saml_config.using
+                        # .. custom_attribute_description: Describes which config is being used: direct (from db), latest,
+                        #     or default.
+                        set_custom_attribute('saml_config.using', 'latest')
+                        conf['saml_sp_configuration'] = latest_config
+                except Exception as e:
+                    log.exception("Error finding latest SAML config for slug %s, site_id %s: %s",
+                                  self.saml_configuration.slug, self.saml_configuration.site_id, e)
+            else:
+                set_custom_attribute('saml_config.using', 'default')
```

Comment:
It seems like the configuration itself is not actually updated in this case? Maybe this logic will need to be updated a bit. However, we may first want to see if we update the code based on the signal before reviewing this again.